communicator/room.go

```diff
@@ -439,11 +439,9 @@ func (r *Room) handlePlaceUnit(client *Client, payload PlaceUnitPayload) {
 		return
 	}
 
-	moved := false
 	for i, existing := range state.UnitsPlacement {
-		if existing.UnitID == placement.UnitID {
-			moved = true
-			state.UnitsPlacement[i] = placement
```

Comment:
I don't see what has changed here. The outcome should be the same (?)

Reply:
Nope. There was a bug.
1. Place unit1 on the gameboard (x = 1, y = 1)
2. Place unit2 on the gameboard (x = 5, y = 5)
3. With unit2 still selected, place it on (1, 1)

Then I expect the unitsPlacement state to be
```
unit2 x=1, y=1
```

But it was:
```
unit2 x=1, y=1
unit2 x=1, y=1
```

It's easy to see that the previous code never removed anything from the gameboard. 
But in this case you had 2 units on the gameboard, and when you moved the one onto the other, the other should disappear, and be replaced.